### PR TITLE
Use source address of request as SDP session and media address

### PR DESCRIPTION
**Pull request by juha-h**
This PR always keep SDP session and media addresses up to date with request source address.

### CODE REVIEW

**Comment by cspiel1** [comment]
I think that `src/sip` should make no assumptions about the structure of the SIP body. So I suppose that SDP has to be built, encoded outside and then passed to function `sip_request()`, or similar.

File: src/sipsess/connect.c [hunk 1]
```diff
@@ -139,18 +139,12 @@ static int invite(struct sipsess *sess)
 			     sess->dlg, 0, sess->auth,
 			     send_handler, invite_resp_handler, sess,
 			     "%b"
-			     "%s%s%s"
-			     "Content-Length: %zu\r\n"
```
[the remark this answers is not in the document]
Comment (juha-h) [reply src/sipsess/connect.c:143]: Yes, I already agreed that this is a bug in the PR.  My solution is keep the existing arguments when sess->sdp is not set and the new ones when it is set (= the common case).

File: src/sip/request.c [hunk 2]
```diff
@@ -176,15 +184,39 @@ static int request(struct sip_request *req, enum sip_transp tp,
...
+
+	/* add Contact header */
 	err |= req->sendh ? req->sendh(tp, &laddr, dst, mb, req->arg) : 0;
 	err |= mbuf_write_mem(mb, mbuf_buf(req->mb), mbuf_get_left(req->mb));
+
+	/* add Content-Length and SDP message body (if present) */
+	sess = req->sdp;
+	if (sess) {
+		sdp_session_set_laddr(sess, &laddr);
+		bb = mbuf_alloc(1024);
```
Comment (sreimers) [inline src/sip/request.c:205]: I think this is not needed since `sdp_encode` allocates mbuf.

Comment (juha-h) [reply src/sip/request.c]: Should the description of `sdp_encode` be fixed?
```
/**
 * Encode an SDP Session into a memory buffer
 *
 * @param mbp   Pointer to allocated memory buffer
 *
```
Based on that I thought that the user needs to allocate the buffer.

**Comment by cspiel1** [comment]
> It would seem logical that `sendh` would handle also the body of the request or there could be more than one handler, for example, message `header handler` and message `body handler`. The latter would be set based on what kind of body type is in question, for example, sdp handler and another one just adds Content-Length header and copies the body text to mb.
> 

This is a good idea. Will have a look on it soon.

**Pull request closed without merging**